### PR TITLE
Update rust.yml (VPN-6400)

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
Ref: https://mozilla-hub.atlassian.net/browse/VPN-6400

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -47,7 +48,7 @@ jobs:
           echo "cache_loaded=false" >> $GITHUB_ENV
         fi
     - name: Git checkout (reattach HEAD to Head Ref)
-      run: git checkout "$(echo ${{ github.head_ref }} | sed -E 's|refs/[a-zA-Z]+/||')"
+      run: run: git checkout "$(echo $HEAD_REF | sed -E 's|refs/[a-zA-Z]+/||')"
```
Comment (oskirby) [inline .github/workflows/rust.yml]: You seem to have `run:` twice on this line which seems like a bug.

Comment (lesleyjanenorton) [reply .github/workflows/rust.yml]: geezus. thanks.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ on:
 env:
   CARGO_TERM_COLOR: always
   RUST_VERSION: 1.67.1
+  HEAD_REF: {{ github.head_ref }}
```
Comment (oskirby) [inline .github/workflows/rust.yml]: I think the syntax for using these variables in github actions is to lead with a dollar sign, so `HEAD_REF: ${{ github.head_ref }}`

Comment (lesleyjanenorton) [reply .github/workflows/rust.yml]: Takes a village, or a Naomi.

**Pull request closed without merging**